sdk/schemaregistry/azure-schemaregistry-avroencoder/azure/schemaregistry/encoder/avroencoder/_schema_registry_avro_encoder.py

```diff
@@ -0,0 +1,285 @@
+         are positional parameters.
+        :paramtype message_type: Callable or None
+        :rtype: MessageType or MessageMetadataDict
+        :raises ~azure.schemaregistry.encoder.avroencoder.exceptions.SchemaParseError:
+            Indicates an issue with parsing schema.
+        :raises ~azure.schemaregistry.encoder.avroencoder.exceptions.SchemaEncodeError:
+            Indicates an issue with encoding data for provided schema.
+        """
+
+        raw_input_schema = schema
+
+        try:
+            schema_fullname = self._avro_encoder.get_schema_fullname(raw_input_schema)
+        except Exception as e:  # pylint:disable=broad-except
+            SchemaParseError(
+                f"Cannot parse schema: {raw_input_schema}", error=e
+            ).raise_with_traceback()
+
+        schema_id = self._get_schema_id(schema_fullname, raw_input_schema)
+        content_type = f"{AVRO_MIME_TYPE}+{schema_id}"
+
+        try:
+            data_bytes = self._avro_encoder.encode(data, raw_input_schema)
+        except Exception as e:  # pylint:disable=broad-except
+            SchemaEncodeError(
+                "Cannot encode value '{}' for schema: {}".format(
+                    data, raw_input_schema
+                ),
+                error=e,
+            ).raise_with_traceback()
+
+        stream = BytesIO()
+
+        stream.write(data_bytes)
+        stream.flush()
+
+        payload = stream.getvalue()
+        stream.close()
+
+        if message_type:
+            try:
+                return message_type.from_message_data(payload, content_type, **kwargs)
+            except AttributeError:
+                try:
+                    return message_type(payload, content_type, **kwargs)
+                except TypeError as e:
+                    SchemaEncodeError(
+                        f"""The data model {str(message_type)} is not a Callable that takes `data`
+                            and `content_type` or a subtype of the MessageType protocol.
+                            If using an Azure SDK model class, please check the README.md for the full list
+                            of supported Azure SDK models and their corresponding versions."""
+                    ).raise_with_traceback()
+
+        return {"data": payload, "content_type": content_type}
+
+    def _convert_preamble_format(
+        self, data, content_type
+    ):  # pylint: disable=no-self-use
+        record_format_identifier = b"\0\0\0\0"
+        if data[0:RECORD_FORMAT_IDENTIFIER_LENGTH] == record_format_identifier:
+            schema_id = data[
+                SCHEMA_ID_START_INDEX : (SCHEMA_ID_START_INDEX + SCHEMA_ID_LENGTH)
+            ].decode("utf-8")
+            content_type = f"{AVRO_MIME_TYPE}+{schema_id}"
+            data = data[DATA_START_INDEX:]
+
+        return data, content_type
+
+    def decode(
+        self,
+        message: Union[MessageType, MessageMetadataDict],
+        *,
+        readers_schema: Optional[str] = None,
```

Comment:
should we mention this in changelog?

Reply:
I thought about this and decided to not include it in the changelog, especially considering that all the other params changed as well and we didn't mention those.

I can put it in if you feel strongly.

Reply:
oh, also just checked the other language changelogs and they don't mention this.

Reply:
I do mention it in JS: https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/schemaregistry/schema-registry-avro/CHANGELOG.md:
> [](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/schemaregistry/schema-registry-avro/CHANGELOG.md#breaking-changes)decodeMessageData now supports decoding using a different but compatible schema

but I am happy to change it if we get consensus. /cc @JoshLove-msft

Reply:
ah gotcha, thanks for the link - I was looking for the specific param name.

Reply:
If we decide to do this, I can just update it in another docs PR later. Going to merge this one for now.